tests/integration-tests/images_factory.py

```diff
@@ -0,0 +1,114 @@
+        match = re.match(r"^{\'image\': \'(.*)\'}", result.split("\n")[6])
+        image_info = json.loads(match.group(1))
+        self._update_image_info(image_info)
+        return result
+
+    def delete(self, force=False):
+        """Delete image."""
+        command = ["pcluster", "delete-image", "--id", self.image_id, "-r", self.region]
+        if force:
+            command.append("--force")
+        if self.keep_logs:
+            command.append("--keep-logs")
+        result = run_command(command).stdout
+        return result
+
+    def describe(self):
+        """Describe image."""
+        logging.info("Describe image {0} in region {1}.".format(self.image_id, self.region))
+        command = ["pcluster", "describe-image", "--id", self.image_id, "-r", self.region]
+        result = run_command(command).stdout
+        # TODO fix it to use json response
+        match = re.match(r"^{\'image\': \'(.*)\'}", result.split("\n")[1])
+        image_info = json.loads(match.group(1))
+        self._update_image_info(image_info)
+        return result
+
+    def get_log(self):
+        """Get image build log."""
+        logging.info("Get image {0} build log.".format(self.image_id))
+        command = ["pcluster", "get-image-log-events", self.image_id, "-r", self.region]
+        result = run_command(command).stdout
+        return result
+
+    def _update_image_info(self, image_info):
+        self.image_tags = image_info.get("image_tags")
+        self.creation_time = image_info.get("creation_time")
+        self.build_log = image_info.get("build_log")
+        self.version = image_info.get("version")
+        self.image_status = image_info.get("imagebuild_status")
+
+
+class ImagesFactory:
+    """Manage creation and destruction of pcluster images."""
+
+    def __init__(self):
+        self.__created_images = {}
+
+    def create_image(self, image: Image):
+        """
+        Create a image with a given config.
+        :param image: image to create.
+        """
+        logging.info("Build image {0} with config {1}".format(image.image_id, image.config_file))
+        result = image.build()
+        self.__created_images[image.image_id] = image
+        return result
+
+    def destroy_image(self, image: Image):
+        """Delete a created image."""
+        logging.info("Delete image {0}".format(image.image_id))
+
+        if image.image_id in self.__created_images:
+            result = image.delete(force=True)
+            del self.__created_images[image.image_id]
+            return result
+
+    def destroy_all_images(self):
+        """Destroy all created images."""
+        logging.debug("Destroying all images")
+        for key in list(self.__created_images.keys()):
+            try:
+                self.destroy_image(self.__created_images[key])
+            except Exception as e:
+                logging.error("Failed when destroying image {0} with exception {1}.".format(key, e))
```

Comment:
When logging a message you should use the %s format because it's the optimized one for logging: https://stackoverflow.com/questions/54367975/python-3-7-logging-f-strings-vs

So I'd use %s for log, and f-string elsewhere, instead of format.

Reply:
use %s for all logs